ProteoformSuiteInternal/ProteoformCommunity.cs

```diff
@@ -96,32 +141,145 @@ private static List<ProteoformRelation> count_nearby_relations(List<ProteoformRe
         public Dictionary<string, List<ProteoformRelation>> relate_ed()
         {
             Dictionary<string, List<ProteoformRelation>> ed_relations = new Dictionary<string, List<ProteoformRelation>>();
-            Parallel.ForEach(decoy_proteoforms, decoys =>
+            if (!Lollipop.limit_TD_BU_theoreticals)
             {
-                ed_relations[decoys.Key] = relate_et(experimental_proteoforms, decoys.Value, ProteoformComparison.ed);
-            });
+                Parallel.ForEach(decoy_proteoforms, decoys =>
+                {
+                    ed_relations[decoys.Key] = relate_et(experimental_proteoforms.Where(p => p.accepted).ToArray(), decoys.Value, ProteoformComparison.ed);
+                });
+            }
+            else
+            {
+                Parallel.ForEach(decoy_proteoforms, decoys =>
+                {
+                    ed_relations[decoys.Key] = relate_et(experimental_proteoforms.Where(p => p.accepted).ToArray(), decoys.Value.Where(t => t.psm_count_BU > 0).ToArray(), ProteoformComparison.ed);
```

Comment:
could simplify this if-else by just making a temporary decoys list with the "? :" operator